### ISSUE TITLE
Clean install (no packages, no fonts) does not work

### ISSUE THREAD

**Issue by iluvatar1** [hlissner/doom-emacs#5841]
### What did you expect to happen?

Installing doom emacs from zero without packages or fonts will just give a working plain emacs 
I am using this since I am creating a virtual machine for students and I need the doom install to be fast. Currently it fails a lot from network errors (but my internet connection is fine) and also takes a lot of time installing packages (it could be nice to install/compile packages in parallel, but that is a suggestion for another issue). 

### What actually happened?

After running the command 
```
rm -rf ~/.emacs.d ~/.doom.d ; git clone --depth 1 https://github.com/hlissner/doom-emacs ~/.emacs.d; time ~/.emacs.d/bin/doom install --no-env --no-install --no-fonts
```
and then running 
```
emacs --debug-init
``` 
I get the error
```
Debugger entered--Lisp error: (file-missing "Cannot open load file" "No such file or directory" "general")
  require(general nil nil)
  load-with-code-conversion("/Users/oquendo/.emacs.d/core/core-keybinds.el" "/Users/oquendo/.emacs.d/core/core-keybinds.el" nil t)
  require(core-keybinds)
  doom-initialize-core-modules()
  (if no-config-p nil (if doom-debug-p (progn (let ((inhibit-message (active-minibuffer-window))) (message #("DOOM Initializing core modules" 0 5 (face font-lock-comment-face)))))) (doom-initialize-core-modules))
  (progn (setq doom-init-modules-p t) (if no-config-p nil (if doom-debug-p (progn (let ((inhibit-message (active-minibuffer-window))) (message #("DOOM Initializing core modules" 0 5 (face font-lock-comment-face)))))) (doom-initialize-core-modules)) (let* ((init-p (and t (condition-case e (let (file-name-handler-alist) (load ... t ...)) ((debug doom-error) (signal ... ...)) ((debug error) (doom--handle-load-error e ... doom-private-dir)))))) (if init-p (progn (if doom-debug-p (progn (let (...) (message ...)))) (maphash (doom-module-loader doom-module-init-file) doom-modules) (doom-run-hooks 'doom-before-init-modules-hook) (if no-config-p nil (maphash (doom-module-loader doom-module-config-file) doom-modules) (doom-run-hooks 'doom-init-modules-hook) (condition-case e (let (file-name-handler-alist) (load ... t ...)) ((debug doom-error) (signal ... ...)) ((debug error) (doom--handle-load-error e ... doom-private-dir))) (if custom-file (progn (load custom-file ... ...))))) nil)))
  (if (or force-p (not doom-init-modules-p)) (progn (setq doom-init-modules-p t) (if no-config-p nil (if doom-debug-p (progn (let ((inhibit-message ...)) (message #("DOOM Initializing core modules" 0 5 ...))))) (doom-initialize-core-modules)) (let* ((init-p (and t (condition-case e (let ... ...) (... ...) (... ...))))) (if init-p (progn (if doom-debug-p (progn (let ... ...))) (maphash (doom-module-loader doom-module-init-file) doom-modules) (doom-run-hooks 'doom-before-init-modules-hook) (if no-config-p nil (maphash (doom-module-loader doom-module-config-file) doom-modules) (doom-run-hooks 'doom-init-modules-hook) (condition-case e (let ... ...) (... ...) (... ...)) (if custom-file (progn ...)))) nil))))
  doom-initialize-modules()
  load-with-code-conversion("/home/oquendo/.emacs.d/init.el" "/home/oquendo/.emacs.d/init.el" t t)
  load("/home/oquendo/.emacs.d/init" noerror nomessage)
  #<subr startup--load-user-init-file>(#f(compiled-function () #<bytecode -0x14b2e89f68f84fb>) #f(compiled-function () #<bytecode -0x1f3c686ddc0d7975>) t)
  apply(#<subr startup--load-user-init-file> (#f(compiled-function () #<bytecode -0x14b2e89f68f84fb>) #f(compiled-function () #<bytecode -0x1f3c686ddc0d7975>) t))
  startup--load-user-init-file(#f(compiled-function () #<bytecode -0x14b2e89f68f84fb>) #f(compiled-function () #<bytecode -0x1f3c686ddc0d7975>) t)
  command-line()
  normal-top-level()
```

### Describe your attempts to resolve the issue

I checked that those files indeed exists. Also I played with the -no-install flags but it takes a lot of time installing packages and fails many times, having to restart the install process consistently.

### Steps to reproduce

1. rm -rf ~/.emacs.d ~/.doom.d ; git clone --depth 1 https://github.com/hlissner/doom-emacs ~/.emacs.d; time ~/.emacs.d/bin/doom install --no-env --no-install --no-fonts
2. emacs 

### System Information

https://pastebin.com/1RKsFDpc

**Comment by hlissner** [comment]
I cannot reproduce these issues on any of my test VMs (macOS Catalina, NixOS, Arch, and Ubuntu 18), on 27.1, 27.2, or 28.0.50 (with or without native-comp). Your errors suggest core packages weren't successfully installed; this sounds like networking issues, which Doom (and Emacs) don't really have much to do with.

Next time you try, pay closer attention to the output of `doom install`. Perhaps there were errors that you missed, or worse, the command didn't complete, but you carried on as normal anyway.

Certainly, Doom's package manager could handle the error better (e.g. retrying a couple times on failure), and [there is ongoing work to implement that](https://github.com/hlissner/doom-emacs/issues/4273), but it wouldn't address the underlying issue.

As there is nothing for Doom to do here (that isn't already covered in #4273), I will close this, but feel free to follow up here. I'm afraid you simply have to try again (maybe later, when your connection is more stable), or double check your VM's networking setup.

**Comment by iluvatar1** [comment]
Hi, thanks for taking the time to look at this. I have tested now in a linux box, with the same error. Is placed in another place, I just ran everything through ssh, so the possible local network errors are not presented. I get the same error, and this is the output from the install command
```
> Installing straight...
  > Cloning use-package...
> Executing 'doom install --no-env --no-install --no-fonts' with Emacs 27.2 at 2021-11-27 10:18:00
  Installing Doom Emacs!

  > Creating .doom.d/
    ✓ Created .doom.d/
  - Creating .doom.d/init.el
  ✓ Done!
  - Creating .doom.d/config.el
  ✓ Done!
  - Creating .doom.d/packages.el
  ✓ Done!
  ! Not generating envvars file, as requested
  ! Not installing plugins, as requested
  Regenerating autoloads files
  > (Re)generating autoloads file...
    > Generating autoloads file...
    > Byte-compiling autoloads file...
    ✓ Generated .local/autoloads.27.2.elc
  Deploying commit-msg and pre-push git hooks
  ✓ Created .git/hooks/commit-msg
  ✓ Created .git/hooks/pre-push
  ✓
  Finished! Doom is ready to go!

  But before you doom yourself, here are some things you should know:

  1. Don't forget to run 'doom sync', then restart Emacs, after modifying
     ~/.doom.d/init.el or ~/.doom.d/packages.el.

     This command ensures needed packages are installed, orphaned packages are
     removed, and your autoloads/cache files are up to date. When in doubt, run
     'doom sync'!

  2. If something goes wrong, run `doom doctor`. It diagnoses common issues with
     your environment and setup, and may offer clues about what is wrong.

  3. Use 'doom upgrade' to update Doom. Doing it any other way will require
     additional steps. Run 'doom help upgrade' to understand those extra steps.

  4. Access Doom's documentation from within Emacs via 'SPC h d h' or 'C-h d h'
     (or 'M-x doom/help')

  Have fun!

  ✓ Finished in 6.9576s
```
And this is the output from doom doctor
```
> Executing 'doom doctor' with Emacs 27.2 at 2021-11-27 10:20:21
  The doctor will see you now...

  > Checking your Emacs version...
  > Checking for Doom's prerequisites...
    x Couldn't find the `rg' binary; this a hard dependecy for Doom, file searches may not work at all
  > Checking for Emacs config conflicts...
  > Checking for great Emacs features...
  > Checking for private config conflicts...
  > Checking for stale elc files...
  > Checking for problematic git global settings...
  > Checking Doom Emacs...
      Error (use-package): Cannot load general
      Eager macro-expansion failure: (void-function general--normalize-extended-def)
    x There was an unexpected error
      Message: Symbol's function definition is void
      Error: (void-function define-key!)
      Backtrace:
        (define-key! [remap apropos] #'consult-apropos [remap bookmark-jump] #'con...
        (progn (define-key! [remap apropos] #'consult-apropos [remap bookmark-jump...
        (condition-case err (progn (define-key! [remap apropos] #'consult-apropos ...
        (condition-case-unless-debug err (progn (define-key! [remap apropos] #'con...
        (progn (condition-case-unless-debug err (progn (define-key! [remap apropos...
        (condition-case err (progn (condition-case-unless-debug err (progn (define...
        (eval-buffer #<buffer  *load*-245001> nil "/home/oquendo/.emacs.d/modules/...
        (load-with-code-conversion "/home/oquendo/.emacs.d/modules/completion/vert...
        (load "/home/oquendo/.emacs.d/modules/completion/vertico/config" t nomessa...
        (let (file-name-handler-alist) (load (expand-file-name file (plist-get pli...
    ! Extended backtrace logged to .emacs.d/.local/doom.error.log
``` 
Is there a way I can put this testing in githhub with some ci to show the error better?  I know you need to reproduce it.

**Comment by hlissner** [comment]
Oh hold up, I just realized you're using `--no-install`. To quote its docs:

```
--no-install
  Don't auto-install packages
```

If you're going to use this flag, you need to run `doom sync` afterwards to eventually install those packages.

**Comment by hlissner** [comment]
If you want a "fast" build for students, consider installing Doom once, somewhere, with byte-compilation disabled (add `(setq straight-disable-compile t)` to `~/.doom.d/init.el` and deploying `~/.emacs.d` to the VMs.

**Comment by hlissner** [comment]
An alternative (with fewer edge cases, but a slightly longer install time) is to copy your fully installed `~/.emacs.d` to your VMs, then run `doom build`. It'll take a while to build all the packages, but it cuts out the time of cloning them all first.

**Comment by iluvatar1** [comment]
Thanks for the suggestions, I will check on them
